c++/src/kj/async.h

```diff
@@ -610,6 +610,11 @@ PromiseFulfillerPair<T> newPromiseAndFulfiller();
 // fulfiller will be of type `PromiseFulfiller<Promise<U>>`.  Thus you pass a `Promise<U>` to the
 // `fulfill()` callback, and the promises are chained.
 
+template <typename T>
+PromiseFulfillerPair<T> newCrossThreadPromiseAndFulfiller();
+// Like `newPromiseAndFulfiller()`, but the fulfiller is allowed to be invoked from any thread,
+// not just the one that called this method.
```

Comment:
Hmm, I don't understand how this could lead to deadlock. For deadlock, you have to have two mutexes locked in opposite orders. We shouldn't be locking any other mutexes while holding an executor lock -- if we are, then that's a bug that needs fixing.

Reply:
As discussed in chat, I was remember something slightly different:
- Thread A locks Mutex A.
- Thread B blocks waiting for Mutex A.
- Thread A waits on Thread B's executor state to change, but it never does.

It looks like this doesn't suffer from the same problem -- the XThreadPaf's disposer does perform a wait on its own thread's executor, but this wait will always succeed because it's just waiting for FULFILLING to become FULFILLED.

So, never mind me.